flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -63,11 +70,19 @@
 
 	private boolean isShutdown;
 
-	public NetworkEnvironment(NetworkEnvironmentConfiguration config, TaskEventPublisher taskEventPublisher) {
+	public NetworkEnvironment(
+			NetworkEnvironmentConfiguration config,
+			TaskEventPublisher taskEventPublisher,
+			TaskManagerMetricGroup taskManagerMetricGroup) {
 		this.config = checkNotNull(config);
 
 		this.networkBufferPool = new NetworkBufferPool(config.numNetworkBuffers(), config.networkBufferSize());
 
+		checkNotNull(taskManagerMetricGroup);
```

Comment:
nit: could we move into a separate static private method?

Reply:
Sorry, what is moved into private method?

Reply:
private static registerNetworkMetrics(ComponentMetricGroup parentMetricGroup) {
  checkNotNull(parentMetricGroup);
  MetricGroup networkGroup = parentMetricGroup.addGroup(METRIC_GROUP_NETWORK);
  networkGroup.<Integer, Gauge<Integer>>gauge(METRIC_TOTAL_MEMORY_SEGMENT, 
  networkBufferPool::getTotalNumberOfMemorySegments);
  networkGroup.<Integer, Gauge<Integer>>gauge(METRIC_AVAILABLE_MEMORY_SEGMENT, 
  networkBufferPool::getNumberOfAvailableMemorySegments);
}